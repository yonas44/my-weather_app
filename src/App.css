* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.App {
  background-image: url(./assets/bg-1.jpg);
  display: flex;
  flex-direction: column;
  background-size: cover;
  gap: 20px;
  color: white;
  padding: 20px 0;
  min-height: 100vh;
  background-color: white;
  margin: 0 auto;
  max-width: 500px;
  overflow: hidden;
  position: relative;
}
