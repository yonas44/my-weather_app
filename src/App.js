import { Route, Routes } from 'react-router';
import AllCountries from './components/AllCountries';
import DetailPage from './components/DetailPage';
import Header from './components/header';
import Continents from './components/Continents';
import Notfound from './components/Notfound';
import './App.css';
import Footer from './components/Footer';

function App() {
  return (
    <div className="App">
      <Header />
      <Routes>
        <Route element={<Continents />} path="/" />
        <Route element={<AllCountries />} exact path="/all-countries" />
        <Route element={<DetailPage />} path="/country" />
        <Route element={<Notfound />} exact path="*" />
      </Routes>
      <Footer />
    </div>
  );
}

export default App;
