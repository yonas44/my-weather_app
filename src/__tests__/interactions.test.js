/* eslint-disable no-restricted-globals */
import { render, screen, fireEvent } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import App from '../App';
import store from '../feature/configureStore';

// Mock API calls
const handlers = [
  rest.get('https://restcountries.com/v3.1/all', (req, res, ctx) => res(
    ctx.json([
      {
        name: {
          common: 'Hello world!',
        },
        flags: {
          svg: '',
        },
        region: 'Africa',
        area: 222,
        timezones: ['UTC'],
      },
    ]),
  )),
  rest.get(
    'https://weatherapi-com.p.rapidapi.com/forecast.json',
    (req, res, ctx) => {
      const country = req.url.searchParams.get('q');
      return res(
        ctx.json(
          [
            {
              forecast: {
                forecastday: [
                  {
                    day: {
                      maxtemp_c: 0,
                      mintemp_c: 0,
                      hum: 0,
                      condition: {},
                    },
                  },
                ],
              },
            },
          ],
          {
            country,
          },
        ),
      );
    },
  ),
];

const server = setupServer(...handlers);

// Enable API mocking before tests.
beforeAll(() => server.listen());

// Reset any runtime request handlers we may add during the tests.
afterEach(() => server.resetHandlers());

// Disable API mocking after the tests are done.
afterAll(() => server.close());

describe('Page navigates', () => {
  test('Component renders in the screen', async () => {
    const history = createMemoryHistory();
    render(
      <Provider store={store}>
        <Router location={history.location} navigator={history}>
          <App />
        </Router>
      </Provider>,
    );

    // Check if AllCountries component renders when a continent is selected.

    fireEvent.click(screen.getByTestId('africa'));
    expect(await screen.findByText(/Size/i)).toBeInTheDocument();
    expect(location.pathname).toBe('/all-countries');

    // Check if DetailPage component renders when a country is selected.
    fireEvent.click(screen.getByText(/Hello/));
    expect(await screen.findByText(/Population/)).toBeInTheDocument();
  });

  test('It matchs the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <App />
          </Router>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
