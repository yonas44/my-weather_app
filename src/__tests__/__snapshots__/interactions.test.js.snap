// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page navigates It matchs the snapshot 1`] = `
<div
  className="App"
>
  <header>
    <div
      style={
        Object {
          "color": "white",
          "textDecoration": "none",
        }
      }
    >
      <svg
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
            "cursor": "pointer",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M328 112L184 256l144 144"
          fill="none"
          strokeLinecap="square"
          strokeMiterlimit="10"
          strokeWidth="48"
        />
      </svg>
    </div>
    <span>
      /country
    </span>
    <div
      className="search-wrapper"
    >
      <div
        className="search-field"
      >
        <input
          id="country-input"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Enter a country name.."
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
  </header>
  <div
    className="detail-page"
  >
    <div
      className="country-header"
    >
      <div
        className="load-wrapper-detail-page"
      >
        <img
          alt="loading"
          src="load.gif"
          style={
            Object {
              "height": "100px",
              "width": "100px",
            }
          }
        />
      </div>
      <img
        alt="flag"
        id="country-header-img"
      />
      <div
        className="header-info"
      >
        <p
          id="country-name"
        >
          Hello world!
        </p>
        <div
          className="weather-info"
        >
          <div
            id="top-section"
          >
            <img
              alt="weather-icon"
            />
            <p />
          </div>
          <div
            id="bottom-section"
          >
            <p>
              °C / °C
            </p>
            <p
              style={
                Object {
                  "fontSize": "0.85rem",
                }
              }
            >
              Humidity: %
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="country-detail"
    >
      <p>
        <strong>
          Capital:
        </strong>
         
      </p>
      <p>
        <strong>
          Population:
        </strong>
         undefined
      </p>
      <p>
        <strong>
          Location:
        </strong>
         °N and °E
      </p>
      <p>
        <strong>
          Subregion:
        </strong>
         
      </p>
    </div>
  </div>
  <footer
    style={
      Object {
        "color": "black",
        "margin": "auto auto 5px auto",
        "textAlign": "center",
      }
    }
  >
    Copyright © 2023 All right reserved | Project by Yonas Tesfu
  </footer>
</div>
`;
